packages/aws-cdk-lib/core/lib/private/synthesis.ts

```diff
@@ -215,25 +219,28 @@ function synthNestedAssemblies(root: IConstruct, options: StageSynthesisOptions)
  * twice for the same construct.
  */
 function invokeAspects(root: IConstruct) {
-  const invokedByPath: { [nodePath: string]: IAspect[] } = { };
+  const invokedByPath: { [nodePath: string]: AspectApplication[] } = { };
 
   let nestedAspectWarning = false;
   recurse(root, []);
 
-  function recurse(construct: IConstruct, inheritedAspects: IAspect[]) {
+  function recurse(construct: IConstruct, inheritedAspects: AspectApplication[]) {
     const node = construct.node;
     const aspects = Aspects.of(construct);
-    const allAspectsHere = [...inheritedAspects ?? [], ...aspects.all];
+
+    let localAspects = aspects.list;
+    const allAspectsHere = sortAspectsByPriority(inheritedAspects, localAspects);
```

Comment:
The original `invokeAspects` function should do everything the same except now also order Aspects by priority when sorting.